ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -205,396 +185,356 @@ - (UIColor *)mapView:(MGLMapView *)mapView fillColorForPolygonAnnotation:(RCTMGL
 {
-    if (annotation.title && annotation.subtitle) {
-
-        NSString *id = [(RCTMGLAnnotation *) annotation id];
+    if (!_onRegionDidChange) { return; }
 
-        NSDictionary *event = @{ @"target": self.reactTag,
-                                 @"src": @{ @"title": annotation.title,
-                                            @"subtitle": annotation.subtitle,
-                                            @"id": id,
-                                            @"latitude": @(annotation.coordinate.latitude),
-                                            @"longitude": @(annotation.coordinate.longitude)} };
-
-        [_eventDispatcher sendInputEventWithName:@"onRightAnnotationTapped" body:event];
-    }
+    CLLocationCoordinate2D region = _map.centerCoordinate;
+    _onRegionDidChange(@{ @"target": self.reactTag,
+                       @"src": @{ @"latitude": @(region.latitude),
+                                  @"longitude": @(region.longitude),
+                                  @"zoomLevel": @(_map.zoomLevel),
+                                  @"direction": @(_map.direction),
+                                  @"pitch": @(0),
+                                  @"animated": @(animated) } });
 }
 
-- (MGLAnnotationImage *)mapView:(MGLMapView *)mapView imageForAnnotation:(id<MGLAnnotation>)annotation
-{
-    NSString *url = [(RCTMGLAnnotation *) annotation annotationImageURL];
-    if (!url) { return nil; }
 
-    CGSize imageSize = [(RCTMGLAnnotation *) annotation annotationImageSize];
-    MGLAnnotationImage *annotationImage = [mapView dequeueReusableAnnotationImageWithIdentifier:url];
-
-    if (!annotationImage) {
-        UIImage *image = nil;
-        if ([url hasPrefix:@"image!"]) {
-            NSString* localImagePath = [url substringFromIndex:6];
-            image = [UIImage imageNamed:localImagePath];
-        } else {
-            image = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:url]]];
-        }
-        UIGraphicsBeginImageContextWithOptions(imageSize, NO, 0.0);
-        [image drawInRect:CGRectMake(0, 0, imageSize.width, imageSize.height)];
-        UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
-        UIGraphicsEndImageContext();
-        annotationImage = [MGLAnnotationImage annotationImageWithImage:newImage reuseIdentifier:url];
-    }
+- (void)mapView:(RCTMapboxGL *)mapView regionWillChangeAnimated:(BOOL)animated
+{
+    if (!_onRegionWillChange) { return; }
 
-    return annotationImage;
+    CLLocationCoordinate2D region = _map.centerCoordinate;
+    _onRegionWillChange(@{ @"target": self.reactTag,
+                           @"src": @{ @"latitude": @(region.latitude),
+                                      @"longitude": @(region.longitude),
+                                      @"zoomLevel": @(_map.zoomLevel),
+                                      @"direction": @(_map.direction),
+                                      @"pitch": @(0),
```

Comment:
Ok. Thanks. Will change. Never used MapKit. Can we create cameras from zoomLevels instead of altitudes in the future, though? Or maybe add pitch to `setCenterCoordinate:zoomLevel:direction:animated:completionHandler:` Otherwise, I can't see an easy way to ease to a known position, heading, zoomLevel and pitch. Am I missing something?